public/service/logger.go

```diff
@@ -2,10 +2,116 @@ package service
 
 import (
 	"fmt"
+	"os"
 
 	"github.com/warpstreamlabs/bento/internal/log"
 )
 
+// airGapLogger adapts a LeveledLogger to implement log.Modular for use in Bento streams.
+type airGapLogger struct {
+	l LeveledLogger
+}
+
+func newAirGapLogger(logger LeveledLogger) log.Modular {
+	return &airGapLogger{l: logger}
+}
+
+func (a *airGapLogger) WithFields(fields map[string]string) log.Modular {
+	if a.l == nil {
+		return nil
+	}
+
+	switch t := a.l.(type) {
+	case interface {
+		WithFields(fields map[string]string) log.Modular
+	}:
+		return &airGapLogger{l: t.WithFields(fields)}
+	}
+
+	return a.clone()
+}
+
+func (a *airGapLogger) With(keyValues ...any) log.Modular {
+	if a.l == nil {
+		return nil
+	}
+
+	switch t := a.l.(type) {
+	case interface {
+		With(keyValues ...any) log.Modular
+	}:
+		return &airGapLogger{l: t.With(keyValues...)}
+	}
+
+	return a.clone()
+}
+
+func (a *airGapLogger) Error(format string, v ...any) {
+	if a.l == nil {
+		return
+	}
+	a.l.Error(format, v...)
+}
+func (a *airGapLogger) Warn(format string, v ...any) {
+	if a.l == nil {
+		return
+	}
+	a.l.Warn(format, v...)
+}
+func (a *airGapLogger) Info(format string, v ...any) {
+	if a.l == nil {
+		return
+	}
+	a.l.Info(format, v...)
```

Comment:
```suggestion
	a.l.Info(fmt.Sprintf(format, v...))
```

I gave it a spin with `builder.SetLogger(slog.Default())` on this branch I would get: 

```
2025/09/13 17:08:11 INFO Input type %v is now active !BADKEY=generate
2025/09/13 17:08:11 INFO Output type %v is now active !BADKEY=stdout
```

but using the old `log.NewBentoLogAdapter(l)` I would get: 

```
2025/09/13 17:18:22 INFO Input type generate is now active path=root.input label=""
2025/09/13 17:18:22 INFO Output type stdout is now active path=root.output label=""
``` 

Not sure if we need to add in fmt.Sprint to all the Warn/Info ... etc. or as another suggestion 👇 - add a new function rather than altering `func (s *StreamBuilder) SetLogger(l *slog.Logger)` in place.